client/src/components/utilities.ts

```diff
@@ -44,7 +44,19 @@ export const api = async <T>(
   }
 }
 
-export const apiDownload = (endpoint: string) => window.open(`/api${endpoint}`)
+export const apiDownload = (endpoint: string) =>
+  new Promise((resolve, reject) => {
+    try {
+      const windowObj = window.open(`/api${endpoint}`)
+      if (windowObj != null) {
+        windowObj.onbeforeunload = () => {
+          resolve('done')
+        }
+      }
+    } catch (err) {
+      reject(err)
```

Comment:
maybe log the error to the console here so we don't need to catch it in the calling functions